automation/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/notebooks/NotebookGCEClusterMonitoringSpec.scala

```diff
@@ -0,0 +1,145 @@
+
+      // Create a cluster
+      withNewRuntime(billingProject) { cluster =>
+        val printStr = "Pause/resume test"
+
+        withWebDriver { implicit driver =>
+          // Create a notebook and execute a cell
+          withNewNotebook(cluster, kernel = Python3) { notebookPage =>
+            notebookPage.executeCell(s"""print("$printStr")""") shouldBe Some(printStr)
+          }
+
+          // Stop the cluster
+          stopAndMonitorRuntime(cluster.googleProject, cluster.clusterName)
+
+          // Start the cluster
+          startAndMonitorRuntime(cluster.googleProject, cluster.clusterName)
+
+          // TODO make tests rename notebooks?
+          val notebookPath = new File("Untitled.ipynb")
+          // Use a longer timeout than default because opening notebooks after resume can be slow
+          withOpenNotebook(cluster, notebookPath, 10.minutes) { notebookPage =>
+            // old output should still exist
+            val firstCell = notebookPage.firstCell
+            notebookPage.cellOutput(firstCell) shouldBe Some(CellOutput(printStr, None))
+            // execute a new cell to make sure the notebook kernel still works
+            notebookPage.runAllCells()
+            notebookPage.executeCell("sum(range(1,10))") shouldBe Some("45")
+          }
+        }
+      }
+    }
+
+    "should update welder on a cluster" taggedAs Retryable in { billingProject =>
+      implicit val ronToken: AuthToken = ronAuthToken
+      val deployWelderLabel = "saturnVersion" // matches deployWelderLabel in Leo reference.conf
+
+      // Create a cluster with welder disabled
+
+      withNewRuntime(
+        billingProject,
+        request = defaultRuntimeRequest.copy(labels = Map(deployWelderLabel -> "true"),
+                                             welderDockerImage = Some(LeonardoConfig.Leonardo.oldWelderDockerImage))
+      ) { cluster =>
+        // Verify welder is running with old version
+        val statusResponse = Welder.getWelderStatus(cluster).attempt.unsafeRunSync()
+        statusResponse.isRight shouldBe true
+        val oldWelderHash = LeonardoConfig.Leonardo.oldWelderDockerImage.split(":")(1)
+        statusResponse.toOption.get.gitHeadCommit should startWith(oldWelderHash)
+
+        // Stop the cluster
+        stopAndMonitorRuntime(cluster.googleProject, cluster.clusterName)
+
+        // Start the cluster
+        startAndMonitorRuntime(cluster.googleProject, cluster.clusterName)
+
+        // Verify welder is now running
+        val curWelderHash = LeonardoConfig.Leonardo.curWelderDockerImage.split(":")(1)
+        val newStatusResponse = Welder.getWelderStatus(cluster).attempt.unsafeRunSync()
+        newStatusResponse.isRight shouldBe true
+        newStatusResponse.toOption.get.gitHeadCommit should startWith(curWelderHash)
+      }
+    }
+
+    // TODO: remove this test once we stop supporting the legacy image
+    "should set environment variables for old image" in { billingProject =>
```

Comment:
this test can be removed since we don't expect legacy images being used with GCE anymore

Reply:
Removed